### DISCUSSION TITLE
Fixed framerate

### DISCUSSION THREAD

**Discussion by underdoeg** [tgfrerer/island#13] [Q&A]
Is there already any mechanism to set a fixed framerate? in our case we are currently aiming for half of vertical sync, so we target something like 30 FPS to ensure a stable render. (and then think about ramping it up to 60 afterwards...)

**Comment by tgfrerer** [comment]
It's important in this case to distinguish between display frame rate and render frame rate. 

In Vulkan, and by extension Island, choosing the right type of swapchain, and PresentMode is how you can influence frame rates. 

In Island, you can select the Presentmode for your window swapchain using setPresentMode, see for example: https://github.com/tgfrerer/island/blob/2b718dc7bdb16fc4835c1b5bf9b8a6b9b1cb2e67/apps/examples/hello_world/hello_world_app/hello_world_app.cpp#L123-L136  

The types of PresentMode available are enumerated here: https://github.com/tgfrerer/island/blob/2b718dc7bdb16fc4835c1b5bf9b8a6b9b1cb2e67/modules/le_renderer/private/le_renderer_types.h#L1025-L1033

Note that Vulkan only requires drivers to implement FIFO mode (which is the default vsynced behaviour) and therefore the fallback in case a selected mode is not available on a particular system.

How each PresentMode is supposed to work in detail is described best in the [vulkan spec](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkPresentModeKHR.html)
